docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+documentation of objects in other projects. The mappings are defined in
+the ``intersphinx_mapping`` dictionary in `docs/conf.py`_, and include
+`python`, `numpy`, `scipy`, `astropy`, `pandas`, `sphinx`, and
+`sphinx_automodapi`.
+
+When we include ``astropy.units.Quantity`` in the documentation, it
+will show up as `astropy.units.Quantity` with a link to Astropy's
+documentation.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
+
+PlasmaPy has certain common substitutions pre-defined so that they can
+be used elsewhere in the documentation. For example, we can write
+``|Quantity|`` instead of ``~astropy.units.Quantity``, and
+``|Particle|`` instead of ``~plasmapy.particles.particle_class.Particle``.
+For an up-to-date list of substitutions, please refer to the
+`docs/common_links.rst`_ file.
+
+Because substitutions are performed when `sphinx` builds the
+documentation, they will not be performed before `help` accesses the
+docstring of an `object`. For example, when ``|Particle|`` is used in
+a docstring, `help` will show it as ``|Particle|`` rather than
+``~plasmapy.particles.particle_class.Particle``. Consequently,
+substitutions should not be used in docstrings when it is important
+that users have quick access to the full path of the `object` (such as
+in the ``See Also`` section).
+
+Writing documentation
+=====================
 
 Docstrings
-==========
+----------
+
+A docstring is a comment at the beginning of a function or another
+object that provides information on how to use that function.
+Docstrings begin with ``r"""`` (required when including backslashes,
+such as using LaTeX code in equations) or ``"""``, and end with
+``"""``.
 
-* All public classes, methods, and functions should have docstrings.
+In order to improve readability and maintain consistency, PlasmaPy uses
+the `numpydoc`_ standard for docstrings.
 
-* PlasmaPy uses the `numpydoc`_ standard for docstrings.
+Example docstring
+~~~~~~~~~~~~~~~~~
 
-* Docstrings must be raw string `literals
-  <https://docs.python.org/3/reference/lexical_analysis.html#literals>`_
-  if they contain backslashes.  A raw string literal is denoted by
-  having an ``r`` immediately precede quotes or triple quotes:
+Here is an example docstring in the `numpydoc`_ format.
 
 .. code-block:: python
+   :caption: Example docstring
 
-   r"""
-   I did not like unstable eigenfunctions at first, but then they
-   grew on me.
-   """
+   import numpy as np
+   import warnings
 
-* Simple private functions may need only a one-line docstring.
+   def subtract(a, b, *, switch_order=False):
+       r"""
+       Return the difference between two integers.
```

Comment:
```suggestion
       Compute the difference between two integers.
```

Just say what the function is doing, we already know it's returning something. This is probably the thing that annoys me the most about our docstrings and I'd like to discourage its usage.